lib/puma/server.rb

```diff
@@ -424,7 +425,7 @@ def process_client(client, buffer)
 
             ThreadPool.clean_thread_locals if clean_thread_locals
 
-            unless client.reset(@status == :run)
+            unless client.reset(@status == :run && @fast_track_ka_timeout)
```

Comment:
OK. I can change this to just call `fast_track_ka_timeout`. The method would be called on every request, so I thought that it would be good to cache the result. However, I'm not 100% sure either that `@max_threads` is always static.